python/cudf/cudf/core/buffer/buffer.py

```diff
@@ -117,11 +117,32 @@ class BufferOwner(Serializable):
     # The set of buffers that point to this owner.
     _slices: weakref.WeakSet[Buffer]
 
-    def __init__(self):
-        raise ValueError(
-            f"do not create a {self.__class__} directly, please "
-            "use the factory function `cudf.core.buffer.as_buffer`"
-        )
+    def __init__(
+        self,
+        *,
+        ptr: int,
+        size: int,
+        owner: object,
+        exposed: bool,
+    ):
+        """Create a new buffer owner.
+
+        Do not use this directly, instead use `_from_device_memory` or
+        `_from_host_memory`.
```

Comment:
If you want to make the constructor generally usable, should we remove this warning? OTOH it feels odd to have a public constructor, then recommend that we use two _internal_ methods for construction. We should probably remove the leading underscores in those two if they're going to continue to be the advised defaults. It looks like there are <10 total instances of both, and they're all in these files, so it shouldn't extend the scope of the PR much.

Reply:
Good point, done